sphinx/util/typing.py

```diff
@@ -147,8 +167,20 @@ def is_system_TypeVar(typ: Any) -> bool:
     return modname == 'typing' and isinstance(typ, TypeVar)
 
 
-def restify(cls: type | None, mode: str = 'fully-qualified-except-typing') -> str:
-    """Convert python class to a reST reference.
+def _is_special_form(obj: Any) -> TypeGuard[_SpecialForm]:
```

Comment:
Special forms are in typeshed so I think I can keep this one. MyPy would complain if the stubs change though.